aiohttp/web_runner.py

```diff
@@ -128,6 +130,31 @@ def name(self) -> str:
             ssl=self._ssl_context, backlog=self._backlog)
 
 
+class NamedPipeSite(BaseSite):
+    __slots__ = ('_path', )
+
+    def __init__(self, runner: 'BaseRunner', path: str, *,
+                 shutdown_timeout: float=60.0) -> None:
+        loop = asyncio.get_event_loop()
+        assert isinstance(loop, asyncio.ProactorEventLoop)
```

Comment:
if-block + raising a `RuntimeError` is probably more appropriate here. `assert`s are ripped off in the optimized mode and we still want to communicate this error to users in a nice way...

Reply:
Okay.
Optimized mode??